pkgs/by-name/ms/msbuild-structured-log-viewer/package.nix

```diff
@@ -0,0 +1,94 @@
+{
+  lib,
+  stdenv,
+  buildDotnetModule,
+  fetchFromGitHub,
+  dotnetCorePackages,
+  autoPatchelfHook,
+  copyDesktopItems,
+  icu,
+  openssl,
+  libkrb5,
+  makeDesktopItem,
+  writeShellScript,
+  nix-update,
+}:
+buildDotnetModule (finalAttrs: rec {
+  pname = "msbuild-structured-log-viewer";
+  version = "2.2.356";
+
+  src = fetchFromGitHub {
+    owner = "KirillOsenkov";
+    repo = "MSBuildStructuredLog";
+    rev = "v${version}";
+    hash = "sha256-ZDBQR0TYB2K6zIfExD1nm0A7q77woL0vLfQvliKxvdA=";
+  };
+
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  projectFile = [ "src/StructuredLogViewer.Avalonia/StructuredLogViewer.Avalonia.csproj" ];
+  nugetDeps = ./deps.nix;
+
+  # HACK: Clear out RuntimeIdentifiers that's set in StructuredLogViewer.Avalonia.csproj, otherwise our --runtime has no effect
+  dotnetFlags = [ "-p:RuntimeIdentifiers=" ];
+
+  nativeBuildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    autoPatchelfHook
+    copyDesktopItems
+  ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isLinux [
+    stdenv.cc.cc.lib
+    icu
+    openssl
+    libkrb5
+  ];
+
+  dontDotnetFixup = true;
+
+  postFixup =
+    ''
+      wrapDotnetProgram $out/lib/${finalAttrs.pname}/StructuredLogViewer.Avalonia $out/bin/${meta.mainProgram}
```

Comment:
I'm renaming the executable name from upstream's `StructuredLogViewer.Avalonia` to `msbuild-structured-log-viewer` here, but is that something we should/can do?